hugolib/scratch.go

```diff
@@ -52,6 +53,39 @@ func (c *Scratch) Get(key string) interface{} {
 	return c.values[key]
 }
 
+// SetToMap stores a value to a map with the given key in the Node context.
+// This map can later be retrieved with GetSortedMap.
+func (c *Scratch) SetToMap(key string, mapKey string, value interface{}) string {
+	_, found := c.values[key]
+    if !found {
+		c.values[key] = make(map[string]interface{})
+	}
+	
+    c.values[key].(map[string]interface{})[mapKey] = value
+    return ""
+}
+
+// GetSortedMap returns a sorted map previously filled with SetToMap
+func (c *Scratch) GetSortedMap(key string) interface{} {
+	if c.values[key] == nil {
+		return nil
+	}
+	
+	var keys []string
+	for mapKey, _ := range c.values[key].(map[string]interface{}) {
+		keys = append(keys, mapKey)
+	}
+
+	sort.Strings(keys)
+	
+	sortedArray := make(map[int]interface{}, len(c.values[key].(map[string]interface{})))
```

Comment:
In Go, since Go 1.1, you have no guarantee about the iteration order of maps. So what you want here is a slice.